src/codec.rs

```diff
@@ -677,6 +686,20 @@ impl Decode for () {
 	}
 }
 
+macro_rules! impl_len {
+	( $( $type:ident< $($g:ident),* > ),* ) => { $(
+		impl<$($g),*> DecodeLength for $type<$($g),*> {
+			fn len(self_encoded: &mut &[u8]) -> Result<usize, Error> {
+				usize::try_from(u32::from(Compact::<u32>::decode(self_encoded)?))
+					.map_err(|_| "can't convert".into())
```

Comment:
Sorry, but can be please get a more meaningful error message? :D

Reply:
Yeah I thought this was the rule (short messages) by seeing some of the other ones :p